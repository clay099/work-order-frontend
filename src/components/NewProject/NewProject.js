import React from "react";
import { useSelector } from "react-redux";
import { Redirect } from "react-router-dom";
import NewProjectForm from "../NewProjectForm/NewProjectForm";

const NewProject = () => {
	// gets user email from store
	const { userType, token } = useSelector((st) => ({
		userType: st.login.user_type,
		token: st.login.token,
	}));

	// if user is not logged and userType is not "user"  you will be redirected to the home page. Otherwise you will go to the new project form
	return userType !== "user" ? (
		<Redirect to="/" />
	) : (
		<NewProjectForm token={token} />
	);
};

export default NewProject;
