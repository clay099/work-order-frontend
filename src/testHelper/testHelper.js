// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
