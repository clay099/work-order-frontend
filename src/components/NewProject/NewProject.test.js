import React from "react";
import NewProject from "./NewProject";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

const initialState = {
	login: { token: "token", user_type: "user" },
};

describe("NewProject Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<NewProject />, {
			initialState,
		});
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<NewProject />, {
			initialState,
		});
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<NewProject />, {
			initialState,
		});
		const value = getByText("Create New Project");
		expect(value).toBeInTheDocument();
	});
});
