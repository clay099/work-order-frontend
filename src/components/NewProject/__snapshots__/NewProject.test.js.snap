// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewProject Component matches snapshot 1`] = `[Function]`;
